rust/lance-index/src/optimize.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub struct MergeIndex {}
+
+pub enum OptimizeAction {
+    Append,
```

Comment:
should we just make `CreateDelta` a mode of `Append`? say `Append(create_new_fragment)`

Reply:
Lemme think about it.  The intent was that one would not merge with the previous index, and the other would merge with the previous index.

Reply:
Probally we just have one option to determine how many previous index to be merged together?